[flaky-test] service/query TestMaxActiveViews/BeginView_succeeds_after_EndView_frees_capacity

```
=== FAIL: service/query TestMaxActiveViews/BeginView_succeeds_after_EndView_frees_capacity (2.24s)
    query_service_test.go:324: generating config and namespaces
    query_service_test.go:324: connection endpoints: [localhost:5433]
2026-02-26 12:49:52.210 UTC 03dd INFO [db-connection] isEndpointReady -> [localhost:5433] Connected to database
    query_service_test.go:324: connection nodes details: localhost:5433
    query_service_test.go:324: [TestMaxActiveViews/BeginView_succeeds_after_EndView_frees_capacity] db name: sc_test_lkwd4wcm7gdjqlxu2rbewao73gt7reknwbnptnjgjco7xxativza
2026-02-26 12:49:52.250 UTC 03de INFO [validator-committer] NewDatabasePool -> DB source: db=sc_test_lkwd4wcm7gdjqlxu2rbewao73gt7reknwbnptnjgjco7xxativza, endpoints=localhost:5433, load_balance=false, tls_mode=
2026-02-26 12:49:52.267 UTC 03df INFO [validator-committer] NewDatabasePool -> DB pool created
2026-02-26 12:49:52.267 UTC 03e0 INFO [validator-committer] newDatabase -> validator persister connected to database at [localhost:5433]
2026-02-26 12:49:52.267 UTC 03e1 INFO [validator-committer] NewValidatorCommitterService -> Initializing new validator committer service.
2026-02-26 12:49:52.267 UTC 03e2 INFO [validator-committer] NewDatabasePool -> DB source: db=sc_test_lkwd4wcm7gdjqlxu2rbewao73gt7reknwbnptnjgjco7xxativza, endpoints=localhost:5433, load_balance=false, tls_mode=
2026-02-26 12:49:52.273 UTC 03e3 INFO [validator-committer] NewDatabasePool -> DB pool created
2026-02-26 12:49:52.273 UTC 03e4 INFO [validator-committer] newDatabase -> validator persister connected to database at [localhost:5433]
2026-02-26 12:49:52.273 UTC 03e5 INFO [validator-committer] newPreparer -> Initializing new preparer
2026-02-26 12:49:52.273 UTC 03e6 INFO [validator-committer] newValidator -> Initializing new validator
2026-02-26 12:49:52.273 UTC 03e7 INFO [validator-committer] newCommitter -> Initializing new committer
2026-02-26 12:49:52.274 UTC 03e8 INFO [grpc-connection] Listener -> Listening on: tcp://127.0.0.1:34011
2026-02-26 12:49:52.274 UTC 03e9 INFO [validator-committer] setupSystemTablesAndNamespaces -> Created tx status table, metadata table, and its methods.
2026-02-26 12:49:52.274 UTC 03ea INFO [validator-committer] Run -> Starting ValidatorCommitterService
2026-02-26 12:49:52.274 UTC 03eb INFO [validator-committer] Run -> Starting 2 workers for the transaction preparer
2026-02-26 12:49:52.274 UTC 03ec INFO [validator-committer] Run -> Starting 2 workers for the transaction validator
2026-02-26 12:49:52.274 UTC 03ed INFO [validator-committer] Run -> Starting 2 workers for the transaction committer
2026-02-26 12:49:52.274 UTC 03ee INFO [validator-committer] run -> Starting transactionCommitter with 2 workers
2026-02-26 12:49:52.274 UTC 03ef INFO [validator-committer] func2 -> Starting transaction batching and forwarding process
2026-02-26 12:49:52.274 UTC 03f0 INFO [validator-committer] run -> Starting Validator with 2 workers
2026-02-26 12:49:52.274 UTC 03f2 INFO [validator-committer] func1 -> Starting Prometheus monitoring server
2026-02-26 12:49:52.274 UTC 03f1 INFO [validator-committer] run -> Starting transaction preparer with 2 workers
2026-02-26 12:49:52.274 UTC 03f3 INFO [grpc-connection] Listener -> Listening on: tcp://127.0.0.1:42267
2026-02-26 12:49:52.274 UTC 03f4 INFO [monitoring] func1 -> Prometheus serving on URL: http://127.0.0.1:42267/metrics
2026-02-26 12:49:52.289 UTC 03f5 INFO [validator-committer] setupSystemTablesAndNamespaces -> namespace _meta: created table and its methods.
2026-02-26 12:49:52.293 UTC 03f6 INFO [validator-committer] setupSystemTablesAndNamespaces -> namespace _config: created table and its methods.
2026-02-26 12:49:52.352 UTC 03f7 INFO [common.tools.configtxgen] completeInitialization -> orderer type: arma
2026-02-26 12:49:52.354 UTC 03f8 INFO [common.tools.configtxgen] GetOutputBlock -> Generating genesis block
2026-02-26 12:49:52.354 UTC 03f9 INFO [common.tools.configtxgen] GetOutputBlock -> Creating application channel genesis block
2026-02-26 12:49:52.357 UTC 03fa INFO [load-gen-client] Run -> Starting workload generation
2026-02-26 12:49:52.357 UTC 03fb INFO [grpc-connection] Listener -> Listening on: tcp://127.0.0.1:33689
2026-02-26 12:49:52.357 UTC 03fc INFO [monitoring] func1 -> Prometheus serving on URL: http://127.0.0.1:33689/metrics
2026-02-26 12:49:52.359 UTC 03fd INFO [validator-committer] setupSystemTablesAndNamespaces -> Created tx status table, metadata table, and its methods.
2026-02-26 12:49:52.361 UTC 03fe INFO [load-gen-client] submitWorkloadSetupTXs -> Submitting TX [e1dca772d9066494b6cfebfdb195723619fc8f6bfbafb029ef7d4ba3a50291c0]. Progress: 0
2026-02-26 12:49:52.365 UTC 03ff INFO [validator-committer] setupSystemTablesAndNamespaces -> namespace _meta: created table and its methods.
2026-02-26 12:49:52.366 UTC 0400 INFO [validator-committer] setupSystemTablesAndNamespaces -> namespace _config: created table and its methods.
2026-02-26 12:49:52.369 UTC 0401 INFO [load-gen-adapters] RunWorkload -> Opening VC stream
2026-02-26 12:49:52.370 UTC 0402 INFO [validator-committer] func2 -> Started a goroutine to send transaction status to the submitter
2026-02-26 12:49:52.370 UTC 0403 INFO [validator-committer] sendTransactionStatus -> Send transaction status
2026-02-26 12:49:52.370 UTC 0404 INFO [validator-committer] func1 -> Started a goroutine to receive and process transactions
2026-02-26 12:49:52.566 UTC 0405 INFO [load-gen-client] submitWorkloadSetupTXs -> TX [b1f754c42668738dd6ffc4f3ed1eeb363812a88d451bb96c88195bfe3228eb82] submitted. Progress: 2
2026-02-26 12:49:52.566 UTC 0406 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:52.566 UTC 0407 INFO [monitoring] func1 -> Prometheus stopped serving
2026-02-26 12:49:52.566 UTC 0408 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:52.566 UTC 0409 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:52.566 UTC 040a ERRO [validator-committer] StartValidateAndCommitStream -> failed to receive transactions from the coordinator: rpc error: code = Canceled desc = context canceled
(1) attached stack trace
  -- stack trace:
  | github.com/hyperledger/fabric-x-committer/service/vc.(*ValidatorCommitterService).receiveTransactions
  | 	/home/runner/work/fabric-x-committer/fabric-x-committer/service/vc/validator_committer_service.go:312
  | github.com/hyperledger/fabric-x-committer/service/vc.(*ValidatorCommitterService).StartValidateAndCommitStream.func1
  | 	/home/runner/work/fabric-x-committer/fabric-x-committer/service/vc/validator_committer_service.go:290
  | golang.org/x/sync/errgroup.(*Group).Go.func1
  | 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93
  | runtime.goexit
  | 	/opt/hostedtoolcache/go/1.25.5/x64/src/runtime/asm_amd64.s:1693
Wraps: (2) failed to receive transactions from the coordinator
Wraps: (3) rpc error: code = Canceled desc = context canceled
Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *status.Error
2026-02-26 12:49:52.566 UTC 040b INFO [load-gen-client] Run -> End workload generation
2026-02-26 12:49:52.567 UTC 040c INFO [validator-committer] NewDatabasePool -> DB source: db=sc_test_6ktdb2mqd5r57zbkayreses4r6cdzw56h66ek2pynrpme5kern5q, endpoints=localhost:5433, load_balance=false, tls_mode=
2026-02-26 12:49:52.572 UTC 040d INFO [validator-committer] NewDatabasePool -> DB pool created
2026-02-26 12:49:52.572 UTC 040e INFO [grpc-connection] Listener -> Listening on: tcp://127.0.0.1:37595
2026-02-26 12:49:52.572 UTC 040f INFO [monitoring] func1 -> Prometheus serving on URL: http://127.0.0.1:37595/metrics
2026-02-26 12:49:52.572 UTC 0410 INFO [grpc-connection] Listener -> Listening on: tcp://127.0.0.1:45917
2026-02-26 12:49:52.573 UTC 0411 INFO [validator-committer] NewDatabasePool -> DB source: db=sc_test_6ktdb2mqd5r57zbkayreses4r6cdzw56h66ek2pynrpme5kern5q, endpoints=localhost:5433, load_balance=false, tls_mode=
2026-02-26 12:49:52.579 UTC 0412 INFO [validator-committer] NewDatabasePool -> DB pool created
2026-02-26 12:49:52.584 UTC 0413 INFO [monitoring] func1 -> Prometheus stopped serving
2026-02-26 12:49:52.584 UTC 0414 INFO [monitoring] func1 -> Prometheus stopped serving
2026-02-26 12:49:52.584 UTC 0415 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:52.584 UTC 0416 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:52.584 UTC 0417 INFO [validator-committer] Run -> ValidatorCommitterService stopped gracefully
2026-02-26 12:49:52.584 UTC 0418 INFO [validator-committer] close -> closing database connection
2026-02-26 12:49:52.596 UTC 0419 INFO [validator-committer] close -> closing database connection
2026-02-26 12:49:52.596 UTC 041a INFO [validator-committer] close -> closing database connection
    query_service_test.go:695: 
        	Error Trace:	/home/runner/work/fabric-x-committer/fabric-x-committer/service/query/query_service_test.go:695
        	            				/opt/hostedtoolcache/go/1.25.5/x64/src/testing/testing.go:1308
        	            				/opt/hostedtoolcache/go/1.25.5/x64/src/testing/testing.go:1572
        	            				/opt/hostedtoolcache/go/1.25.5/x64/src/testing/testing.go:1928
        	            				/opt/hostedtoolcache/go/1.25.5/x64/src/runtime/asm_amd64.s:1693
        	Error:      	Received unexpected error:
        	            	rpc error: code = FailedPrecondition desc = invalid or stale view
        	Test:       	TestMaxActiveViews/BeginView_succeeds_after_EndView_frees_capacity
        	Messages:   	view created in /home/runner/work/fabric-x-committer/fabric-x-committer/service/query/query_service_test.go:329
2026-02-26 12:49:54.387 UTC 0458 INFO [monitoring] func1 -> Prometheus stopped serving
2026-02-26 12:49:54.387 UTC 0459 INFO [grpc-connection] ] -> Closing 1 connections.
2026-02-26 12:49:54.414 UTC 045a INFO [validator-committer] close -> closing database connection
2026-02-26 12:49:54.415 UTC 045b INFO [validator-committer] close -> closing database connection
    --- FAIL: TestMaxActiveViews/BeginView_succeeds_after_EndView_frees_capacity (2.24s)

=== FAIL: service/query TestMaxActiveViews (0.00s)
```

--- a/service/query/query_service_test.go
+++ b/service/query/query_service_test.go
@@ -10,8 +10,6 @@ import (
 	"context"
 	"encoding/base64"
 	"fmt"
-	"runtime"
-	"slices"
 	"strings"
 	"testing"
 	"time"
@@ -38,12 +36,11 @@ import (
 
 type (
 	queryServiceTestEnv struct {
-		config        *Config
-		qs            *Service
-		ns            []string
-		clientConn    committerpb.QueryServiceClient
-		pool          *pgxpool.Pool
-		disabledViews []string
+		config     *Config
+		qs         *Service
+		ns         []string
+		clientConn committerpb.QueryServiceClient
+		pool       *pgxpool.Pool
 	}
 
 	queryServiceTestOpts struct {
@@ -683,17 +680,10 @@ func (q *queryServiceTestEnv) beginView(
 	require.NoError(t, err)
 	require.NotNil(t, view)
 	require.NotEmpty(t, view.Id)
-	_, file, line, _ := runtime.Caller(1)
-	t.Cleanup(func() {
-		if slices.Contains(q.disabledViews, view.Id) {
-			return
-		}
-		//nolint:usetesting // t.Context() is dead at cleanup.
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		t.Cleanup(cancel)
-		_, err = client.EndView(ctx, view)
-		require.NoErrorf(t, connection.FilterStreamRPCError(err), "view created in %s:%d", file, line)
-	})
+
+	// beginView creates a view and returns it. No explicit cleanup is registered because views are
+	// automatically cleaned up when the service context is cancelled (via context.AfterFunc in makeView).
+	// The service context is derived from t.Context(), which Go cancels before t.Cleanup runs.
 	return view
 }
 
@@ -705,7 +695,6 @@ func (q *queryServiceTestEnv) endView(
 	t.Helper()
 	_, err := client.EndView(t.Context(), view)
 	require.NoError(t, err)
-	q.disabledViews = append(q.disabledViews, view.Id)
 }
 
 func (q *queryServiceTestEnv) insertSampleKeysValueItems(t *testing.T) []*items {

